packages/xstate-angular/src/useActor.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  Actor,
+  ActorOptions,
+  AnyActorLogic,
+  Snapshot,
+  SnapshotFrom,
+  isActorLogic
+} from 'xstate';
+import {
+  Injectable,
+  Signal,
+  signal,
+  Type,
+  WritableSignal
+} from '@angular/core';
+import { useSelector } from './useSelector.ts';
+import { useActorRef } from './useActorRef.ts';
+
+export interface UseActorConfig {
+  providedIn: 'root';
+}
```

Comment:
Could you ELI5 this thing?

Reply:
Could this become part of the `options` parameter (it could be typed as `ActorOptions<TLogic> & { providedIn?: string }`). I'm not a particular fan of making this an optional prepended parameter - it makes consuming those internally more complicated.

Reply:
possible yes, just using it wouldn't be as nice, because you can easily distinguish right now based on the usage where it is provided, e.g. 
`useActor({providedIn: 'root'}, toggleMachine)` => globally provided
`useActor(toggleMachine)` => locally provided

If I understand you correctly yours would look kinda like this,
`useActor({...toggleMachine, providedIn: 'root'})` => globally provided
`useActor(toggleMachine)` => locally provided

Reply:
I was thinking about:
```ts
useActor(toggleMachine, { providedIn: 'root' }) => globally provided
useActor(toggleMachine) => locally provided
useActor(toggleMachine, { input: 42, providedIn: 'root' }) => globally provided, with some other options
```

Reply:
I like that, I will make changes accordingly! Thanks for the input